src/ui/server/controllers/insurance/business/company-details/index.ts

```diff
@@ -142,9 +150,57 @@ const postCompaniesHouseSearch = async (req: Request, res: Response) => {
   }
 };
 
+/**
+ * saves and goes back to all sections from company details page unless there are database errors
+ * @param req
+ * @param res
+ * @param {Express.Request} Express request
+ * @param {Express.Response} Express response
+ * @returns {Express.Response.redirect} redirects to all sections page on success
+ */
+const postCompanyDetailsSaveAndBack = async (req: Request, res: Response) => {
+  try {
+    const { application } = res.locals;
+
+    const { referenceNumber } = req.params;
+
+    if (!application) {
+      return res.redirect(ROUTES.PROBLEM_WITH_SERVICE);
+    }
+
+    const { body } = req;
+    // runs companiesHouse validation and api call first for companiesHouse input
+    const response = await companiesHouseSearch(body);
+    let { validationErrors } = response;
+    const { company } = response;
+
+    // run validation on other fields on page
+    validationErrors = companyDetailsValidation(body, validationErrors);
+
+    // body for update containing companies house info and request body
+    const updateBody = {
+      ...body,
+      ...company,
```

Comment:
I'm wondering if this is functionality we want - if an Exporter enters a CH number, clicks "save and back", and we save the company - we would be saving a company that the Exporter hasn't reviewed - might be a typo in the CH number. Can we check with Tim/Tinu on if this is desired behaviour?

Reply:
discussed with Tinu.  We can do this post MVP as check your answers is there for them to double check